### PR TITLE
fixed inconsistency in allowed precision

### PR DESCRIPTION
**Pull request by RohitRathore1**
It resolves issue #40538 .

### CODE REVIEW

**Comment by dkarrasch** [comment]
The fix seems to be more complicated? This causes many errors.

**Comment by RohitRathore1** [comment]
@dkarrasch now, all have been cleared.

**Comment by dkarrasch** [comment]
Should we have tests that check the behavior for `setprecision(1)`? I'm not an expert here, it's just that we have moved the "corner case" and should make sure it works as expected.

**Comment by RohitRathore1** [comment]
> Should we have tests that check the behavior for `setprecision(1)`? I'm not an expert here, it's just that we have moved the "corner case" and should make sure it works as expected.
I had checked it was working as expected.

**Comment by RohitRathore1** [comment]
> Should we have tests that check the behavior for `setprecision(1)`? I'm not an expert here, it's just that we have moved the "corner case" and should make sure it works as expected.

Actually, it's working as expected. I had checked.

**Comment by ViralBShah** [comment]
We should have someone who has worked on mpfr.jl review this before merging.

File: test/mpfr.jl
Comment (stevengj) [inline test/mpfr.jl:478]: Probably we should explicitly test that `setprecision(1)` is allowed and results in 1-bit BigFloats.

Comment (RohitRathore1) [reply test/mpfr.jl:478]: Sorry I did not get 😔😔

Comment (vtjnash) [reply test/mpfr.jl:478]:
```suggestion
    @test precision(setprecision(1) do; BigFloat(23); end) == 1  # minimum-precision
    @test_throws DomainError setprecision(0)
```

**Review by stevengj — APPROVED** [review]

**Comment by stevengj** [comment]
As it says in the linked issue, the MPFR documentation is very clear that the minimum precision (since 2016) is 1.

**Comment by RohitRathore1** [comment]
> As it says in the linked issue, the MPFR code is very clear that the minimum precision (since 2016) is 1, so that PR seems correct to me.

Thank you @stevengj I am waiting for it to be merged and I am very excited. Thank you!

**Pull request merged**
